### ISSUE TITLE
Panels not displayed on cinnamon 2.6 if xdg-screensaver command is used

### ISSUE THREAD

**Issue by margamanterola** [linuxmint/Cinnamon#4491]
Hi,

cinnamon used to work perfectly fine when using a remote desktop (chrome remote desktop, display is at :20 instead of :0) using version 2.2.  After upgrading to version 2.6 (running on Ubuntu Trusty), the panels have stopped displaying when cinnamon starts.

Everything else works fine: it's possible to right click -> open a terminal -> run cinnamon settings and everything reacts as expected. Except that the panel/s is/are not drawn.

Adding a panel from within the session (if there is an empty spot) is possible, and in that case the panel is displayed and works perfectly fine. However, as soon as cinnamon is restarted (by restarting the session or by running cinnamon --replace) the newly added panel goes missing again.

This happens both in the case of a machine that is already running a local X and a machine without a local X.  The problem is not about "multi-display" but rather "non :0 display" at the start of the session.

Inspecting with LookingGlass shows the panel objects to be there, with the right properties in their actors, but there's no way of getting them to be drawn.

For one of the debugging sessions I checked (with the help of dalcde):

Main.panel.actor.x = 0 
Main.panel.actor.y = 0 
Main.panel.actor.height = 25
Main.panel.actor.width = 1920
Main.panel.actor.visible = true
Main.panel.actor.get_parent() = CinnamonGenericContainer "uiGroup"

Calling functions like show_all() or raise_top() made no difference.

At the javascript level everything looks fine. I suspect the problem is at the clutter level. And it's notable that it only happens on session start, but works fine mid-session.

Given the general migration towards systemd and that this is trusty, it is possible that there's something misconfigured there, but I don't think that's the case, because it works fine mid-session.

**Comment by margamanterola** [comment]
An interesting observation:

In the above report was starting the session with:
/usr/sbin/lightdm-session 'cinnamon-session --session=cinnamon2d'

If instead, the session is started with:
/usr/sbin/lightdm-session /usr/bin/cinnamon2d

The panels are shown correctly, although other stuff that is done by the session, like launching nemo, cinnamon-settings-daemon, etc. is of course not there.

**Comment by margamanterola** [comment]
I finally figured out what was happening.  In my autostart programs, I had one that was intended to inhibit the screensaver, as we don't want a screensaver on a remote session.  This script was using xdg-screensaver, which is not aware of cinnamon-screensaver being special, and thus when issuing the command to inhibit the screensaver, it was actually starting a new gnome-screensaver.

Apparently, gnome-screensaver always says "True" to the "GetActive" call, and as a consequence, LayoutManager's ScreenSaverProxy was stating that the screensaver was "Active"... Possibly because it was asking the first screensaver on the bus.

In any case, I fixed my personal issue by improving the autostart script to special case cinnamon and use the appropriate dbus call instead of the xdg-screensaver command.  What would still need to get fixed:
1) Add cinnamon support to xdg-screensaver
2) Make sure that the ScreenSaverProxy of the LayoutManager talks to the cinnamon screensaver on the bus, and not to just any screensaver.
